core/blockchain_reader.go

```diff
@@ -362,6 +365,24 @@ func (bc *BlockChain) StateAt(startAtRoot common.Hash, startAtBlockHash common.H
 	return sdb, err
 }
 
+// StorageTrieAt returns a target block storage trie.
+func (bc *BlockChain) StorageTrieAt(startAtRoot common.Hash, addr common.Address) (state.Trie, error) {
+	if bc.snaps == nil {
+		return nil, fmt.Errorf("StorageTrieAt snaps is nil, root: %v, addr: %v", startAtRoot, addr)
+	}
+	snap := bc.snaps.Snapshot(startAtRoot)
+	if snap == nil {
+		return nil, fmt.Errorf("cannot open snap to find target storage trie, root: %v, addr: %v", startAtRoot, addr)
+	}
+
+	account, err := snap.Account(crypto.Keccak256Hash(addr[:]))
+	if err != nil {
+		return nil, err
+	}
+
+	return bc.stateCache.OpenStorageTrie(startAtRoot, addr, common.BytesToHash(account.Root))
```

Comment:
Do we need to add a check if `account.Root` is nil? If it's nil, we will just return `(nil,nil)` or `(nil, error)`.

Reply:
I think it is better to return an empty trie other than nil, and return err is unnecessary.